agent/consul/server.go

```diff
@@ -1010,20 +1010,56 @@ func (s *Server) WANMembers() []serf.Member {
 }
 
 // RemoveFailedNode is used to remove a failed node from the cluster
-func (s *Server) RemoveFailedNode(node string) error {
-	if err := s.serfLAN.RemoveFailedNode(node); err != nil {
+// We must remove a node from both serf LAN & WAN if present and also
+// check for the prune flag to see if it should be completely deleted
+func (s *Server) RemoveFailedNode(node string, prune bool) error {
+	if prune {
+		s.removeFailedNodePrune(node)
+	}
+
+	lanRemover := s.serfLAN.RemoveFailedNode
```

Comment:
Why did you introduce these intermediate function indirections that appear to be only used once each.

Reply:
I was looking for the easiest way to understand what is happening here, without having layers of `if` statements.